src/ServicePulse.Host/app/js/views/redirect/service.js

```diff
@@ -0,0 +1,105 @@
+﻿; (function (window, angular, undefined) {
+    'use strict';
+    
+    function service($http, $timeout, $q, $rootScope, $interval, scConfig, uri, notifications, notifyService) {
+        var notifier = notifyService();
+
+        var redirects = {
+            total :0,
+            data: []
+        };
+
+        function getData() {
+            var url = uri.join(scConfig.service_control_url, 'redirects');
+            return $http.get(url).then(function (response) {
+                notifier.notify('RedirectsUpdated', { total: response.headers('Total-Count'), data: response.data });
+                notifier.notify('RedirectMessageCountUpdated', response.headers('Total-Count'));
+            });
+        }
+
+        var redirectsUpdatedTimer = $interval(function () {
+            getData();
+        }, 15000);
+
+        // Cancel interval on page changes
+        $rootScope.$on('$destroy', function () {
+            if (angular.isDefined(redirectsUpdatedTimer)) {
+                $interval.cancel(redirectsUpdatedTimer);
+                redirectsUpdatedTimer = undefined;
+            }
+        });
+
+        notifier.subscribe($rootScope, function (event, response) {
+            redirects.data = response.data;
+            redirects.total = response.total;
+        }, 'RedirectsUpdated');
+
+        function sendPromise(url, method, data, success, error) {
+
+            var defer = $q.defer();
+
+            success = success || 'success';
+            error = error || 'error';
+
+            $http({
+                    url: url,
+                    data: data,
+                    method: method
+                })
+                .then(function(response) {
+                        defer.resolve({ message: success, status: response.status });
+                    }, function(response) {
+                        defer.reject({ message: error + ':' + response.statusText, status: response.status, statusText: response.statusText });
+                    }
+                );
+
+            return defer.promise;
+        }
+
+        getData();
+
+        return {
+            createRedirect: function (sourceEndpoint, targetEndpoint, success, error) {
+                var url = uri.join(scConfig.service_control_url, 'redirects');
+                var promise = sendPromise(url, 'POST', { "fromphysicaladdress": sourceEndpoint, "tophysicaladdress": targetEndpoint }, success, error);
+                promise.then(function() {
+                    getData();
```

Comment:
`getData();` returns a promise, so this call could be replaced with:

```Javascript
return promise.then(getData);
```

Which then keeps the promise chain intact for future use or stacktraces.